src/Psl/Hash/Hmac/Context.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Hash\Hmac;
+
+use HashContext;
+use Psl;
+use Psl\Arr;
+
+use function hash_final;
+use function hash_init;
+use function hash_update;
+
+/**
+ * Incremental hashing context.
+ *
+ * Example:
+ *
+ *      Hmac\Context::create('ripemd160', 'secret')
+ *          ->update('The quick brown fox ')
+ *          ->update('jumped over the lazy dog.')
+ *          ->finalize()
+ *      => Str("b8e7ae12510bdfb1812e463a7f086122cf37e4f7")
+ *
+ * @psalm-immutable
+ */
+final class Context
+{
+    private HashContext $internalContext;
+
+    private function __construct(HashContext $internal_context)
+    {
+        $this->internalContext = $internal_context;
+    }
+
+    /**
+     * Initialize an incremental hashing context.
+     *
+     * @throws Psl\Exception\InvariantViolationException If the given algorithm is unsupported.
+     */
+    public static function create(string $algorithm, string $key): Context
```

Comment:
Would it make sense to create 1 hashing context with 2 named constructors?

- One for regular hashes: `Context::forAlgorithm(...)`
- One for hmac hashes: `Context::hmac()`

The rest seems similar.

Reply:
Maybe best to switch algorithm and key in here as well for consistency?

Reply:
not sure 🤔 it made sense to have it in a separate `Hmac` namespace with other hmac functions

Reply:
`Hmac\Context` has been merged with `Hash\Context` as suggested 👍🏼